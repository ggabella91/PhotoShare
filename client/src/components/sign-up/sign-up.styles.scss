.link-container {
  margin-top: 20px;
}

.link {
  font-size: 14px;
  text-align: center;
  vertical-align: middle;
  font-family: 'Roboto', sans-serif;
  color: black;
  text-decoration: none;
  cursor: pointer;

  &:hover,
  &:active {
    text-decoration: none;
    color: black;
  }
}

span {
  font-size: 20px;
  text-align: center;
  display: table-cell;
  vertical-align: middle;
  margin: auto;
  font-family: 'Roboto', sans-serif;
}

.button {
  border-radius: 15px;
}
