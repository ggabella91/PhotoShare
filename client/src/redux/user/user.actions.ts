import {
  UserActions,
  UserActionTypes,
  UserSignUp,
  UserSignIn,
  User,
  FieldsToUpdate,
  OtherUserRequest,
  ChangePassword,
  ResetPassword,
  Error,
} from './user.types';

export const signUpStart = ({
  username,
  name,
  email,
  password,
  passwordConfirm,
}: UserSignUp): UserActionTypes => ({
  type: UserActions.SIGN_UP_START,
  payload: { username, name, email, password, passwordConfirm },
});

export const signUpSuccess = (user: User): UserActionTypes => ({
  type: UserActions.SIGN_UP_SUCCESS,
  payload: user,
});

export const signUpFailure = (error: Error): UserActionTypes => ({
  type: UserActions.SIGN_UP_FAILURE,
  payload: error,
});

export const checkUserSession = (): UserActionTypes => ({
  type: UserActions.CHECK_USER_SESSION,
  payload: null,
});

export const setCurrentUser = (user: User | null): UserActionTypes => ({
  type: UserActions.SET_CURRENT_USER,
  payload: user,
});

export const signInStart = ({
  email,
  password,
}: UserSignIn): UserActionTypes => ({
  type: UserActions.SIGN_IN_START,
  payload: {
    email,
    password,
  },
});

export const signInSuccess = (user: User): UserActionTypes => ({
  type: UserActions.SIGN_IN_SUCCESS,
  payload: user,
});

export const signInFailure = (error: Error): UserActionTypes => ({
  type: UserActions.SIGN_IN_FAILURE,
  payload: error,
});

export const signOutStart = (): UserActionTypes => ({
  type: UserActions.SIGN_OUT_START,
  payload: null,
});

export const signOutSuccess = (): UserActionTypes => ({
  type: UserActions.SIGN_OUT_SUCCESS,
  payload: null,
});

export const signOutFailure = (error: Error): UserActionTypes => ({
  type: UserActions.SIGN_OUT_FAILURE,
  payload: error,
});

export const changeInfoStart = (
  fieldsToUpdate: FieldsToUpdate
): UserActionTypes => ({
  type: UserActions.CHANGE_INFO_START,
  payload: fieldsToUpdate,
});

export const changeInfoSuccess = (user: User): UserActionTypes => ({
  type: UserActions.CHANGE_INFO_SUCCESS,
  payload: user,
});

export const changeInfoFailure = (error: Error): UserActionTypes => ({
  type: UserActions.CHANGE_INFO_FAILURE,
  payload: error,
});

export const changePasswordStart = ({
  passwordCurrent,
  password,
  passwordConfirm,
}: ChangePassword): UserActionTypes => ({
  type: UserActions.CHANGE_PASSWORD_START,
  payload: { passwordCurrent, password, passwordConfirm },
});

export const changePasswordSuccess = (message: string): UserActionTypes => ({
  type: UserActions.CHANGE_PASSWORD_SUCCESS,
  payload: message,
});

export const changePasswordFailure = (error: Error): UserActionTypes => ({
  type: UserActions.CHANGE_PASSWORD_FAILURE,
  payload: error,
});

export const forgotPasswordStart = (email: string): UserActionTypes => ({
  type: UserActions.FORGOT_PASSWORD_START,
  payload: email,
});

export const forgotPasswordSuccess = (message: string): UserActionTypes => ({
  type: UserActions.FORGOT_PASSWORD_SUCCESS,
  payload: message,
});

export const forgotPasswordFailure = (error: Error): UserActionTypes => ({
  type: UserActions.FORGOT_PASSWORD_FAILURE,
  payload: error,
});

export const resetPasswordStart = ({
  password,
  passwordConfirm,
  token,
}: ResetPassword): UserActionTypes => ({
  type: UserActions.RESET_PASSWORD_START,
  payload: { password, passwordConfirm, token },
});

export const resetPasswordSuccess = (message: string): UserActionTypes => ({
  type: UserActions.RESET_PASSWORD_SUCCESS,
  payload: message,
});

export const resetPasswordFailure = (error: Error): UserActionTypes => ({
  type: UserActions.RESET_PASSWORD_FAILURE,
  payload: error,
});

export const deleteAccountStart = (): UserActionTypes => ({
  type: UserActions.DELETE_ACCOUNT_START,
  payload: null,
});

export const deleteAccountSuccess = (message: string): UserActionTypes => ({
  type: UserActions.DELETE_ACCOUNT_SUCCESS,
  payload: message,
});

export const deleteAccountFailure = (error: Error): UserActionTypes => ({
  type: UserActions.DELETE_ACCOUNT_FAILURE,
  payload: error,
});

export const clearInfoStatuses = (): UserActionTypes => ({
  type: UserActions.CLEAR_INFO_STATUSES,
  payload: null,
});

export const clearPasswordStatuses = (): UserActionTypes => ({
  type: UserActions.CLEAR_PASSWORD_STATUSES,
  payload: null,
});

export const getOtherUserStart = (
  otherUserReq: OtherUserRequest
): UserActionTypes => ({
  type: UserActions.GET_OTHER_USER_START,
  payload: otherUserReq,
});

export const getOtherUserSuccess = (user: User): UserActionTypes => ({
  type: UserActions.GET_OTHER_USER_SUCCESS,
  payload: user,
});

export const getFollowersSuccess = (user: User): UserActionTypes => ({
  type: UserActions.GET_FOLLOWERS_INFO_SUCCESS,
  payload: user,
});

export const getFollowingSuccess = (user: User): UserActionTypes => ({
  type: UserActions.GET_FOLLOWING_INFO_SUCCESS,
  payload: user,
});

export const getPostReactorsSuccess = (user: User): UserActionTypes => ({
  type: UserActions.GET_POST_REACTOR_SUCCESS,
  payload: user,
});

export const getOtherUserFailure = (error: Error): UserActionTypes => ({
  type: UserActions.GET_OTHER_USER_FAILURE,
  payload: error,
});

export const getUserSuggestionsStart = (match: string): UserActionTypes => ({
  type: UserActions.GET_USER_SUGGESTIONS_START,
  payload: match,
});

export const getUserSuggestionsSuccess = (users: User[]): UserActionTypes => ({
  type: UserActions.GET_USER_SUGGESTIONS_SUCCESS,
  payload: users,
});

export const getUserSuggestionsFailure = (error: Error): UserActionTypes => ({
  type: UserActions.GET_USER_SUGGESTIONS_FAILURE,
  payload: error,
});

export const clearUserSuggestions = (): UserActionTypes => ({
  type: UserActions.CLEAR_USER_SUGGESTIONS,
  payload: null,
});

export const clearFollowersAndFollowing = (): UserActionTypes => ({
  type: UserActions.CLEAR_FOLLOWERS_AND_FOLLOWING,
  payload: null,
});

export const setIsCurrentUserProfilePage = (
  isCurrentUserProfilePage: boolean
): UserActionTypes => ({
  type: UserActions.SET_IS_CURRENT_USER_PROFILE_PAGE,
  payload: isCurrentUserProfilePage,
});

export const getFeedPostReactorsSuccess = (user: User): UserActionTypes => ({
  type: UserActions.GET_FEED_POST_REACTOR_SUCCESS,
  payload: user,
});

export const getConversationUserSuccess = (user: User): UserActionTypes => ({
  type: UserActions.GET_CONVERSATION_USER_SUCCESS,
  payload: user,
});

export const getNotificationUserSuccess = (user: User): UserActionTypes => ({
  type: UserActions.GET_NOTIFICATION_USER_SUCCESS,
  payload: user,
});

export const clearConversationUsers = (): UserActionTypes => ({
  type: UserActions.CLEAR_CONVERSATION_USERS,
});
