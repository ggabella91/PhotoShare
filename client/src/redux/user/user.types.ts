export enum UserActions {
  CHECK_USER_SESSION = 'CHECK_USER_SESSION',
  SET_CURRENT_USER = 'SET_CURRENT_USER',
  SIGN_UP_START = 'SIGN_UP_START',
  SIGN_UP_SUCCESS = 'SIGN_UP_SUCCESS',
  SIGN_UP_FAILURE = 'SIGN_UP_FAILURE',
  SIGN_IN_START = 'SIGN_IN_START',
  SIGN_IN_SUCCESS = 'SIGN_IN_SUCCESS',
  SIGN_IN_FAILURE = 'SIGN_IN_FAILURE',
  SIGN_OUT_START = 'SIGN_OUT_START',
  SIGN_OUT_SUCCESS = 'SIGN_OUT_SUCCESS',
  SIGN_OUT_FAILURE = 'SIGN_OUT_FAILURE',
  CHANGE_INFO_START = 'CHANGE_INFO_START',
  CHANGE_INFO_SUCCESS = 'CHANGE_INFO_SUCCESS',
  CHANGE_INFO_FAILURE = 'CHANGE_INFO_FAILURE',
  CHANGE_PASSWORD_START = 'CHANGE_PASSWORD_START',
  CHANGE_PASSWORD_SUCCESS = 'CHANGE_PASSWORD_SUCCESS',
  CHANGE_PASSWORD_FAILURE = 'CHANGE_PASSWORD_FAILURE',
  FORGOT_PASSWORD_START = 'FORGOT_PASSWORD_START',
  FORGOT_PASSWORD_SUCCESS = 'FORGOT_PASSWORD_SUCCESS',
  FORGOT_PASSWORD_FAILURE = 'FORGOT_PASSWORD_FAILURE',
  RESET_PASSWORD_START = 'RESET_PASSWORD_START',
  RESET_PASSWORD_SUCCESS = 'RESET_PASSWORD_SUCCESS',
  RESET_PASSWORD_FAILURE = 'RESET_PASSWORD_FAILURE',
  DELETE_ACCOUNT_START = 'DELETE_ACCOUNT_START',
  DELETE_ACCOUNT_SUCCESS = 'DELETE_ACCOUNT_SUCCESS',
  DELETE_ACCOUNT_FAILURE = 'DELETE_ACCOUNT_FAILURE',
  CLEAR_INFO_STATUSES = 'CLEAR_INFO_STATUSES',
  CLEAR_PASSWORD_STATUSES = 'CLEAR_PASSWORD_STATUSES',
  GET_OTHER_USER_START = 'GET_OTHER_USER_START',
  GET_OTHER_USER_SUCCESS = 'GET_OTHER_USER_SUCCESS',
  GET_OTHER_USER_FAILURE = 'GET_OTHER_USER_FAILURE',
  CLEAR_OTHER_USER_INFO = 'CLEAR_OTHER_USER_INFO',
}

export interface UserSignUp {
  name: string;
  email: string;
  password: string;
  passwordConfirm: string;
}

export interface UserSignIn {
  email: string;
  password: string;
}

export interface User {
  id: string;
  name: string;
  email: string;
  username: string;
  photo?: string;
  bio?: string;
}

export interface FieldsToUpdate {
  name?: string;
  email?: string;
  username?: string;
  bio?: string;
  photo?: string;
}

export interface ChangePassword {
  passwordCurrent: string;
  password: string;
  passwordConfirm: string;
}

export interface ResetPassword {
  password: string;
  passwordConfirm: string;
  token: string;
}

export interface Error {
  statusCode: number;
  message: string;
}

export type UserPayload =
  | UserSignUp
  | UserSignIn
  | User
  | ChangePassword
  | ResetPassword
  | Error
  | null;

export interface UserState {
  currentUser: User | null;
  otherUser: User | null;
  otherUserError: Error | null;
  signUpError: Error | null;
  signInOrOutError: Error | null;
  changeInfoConfirm: null | string;
  changeInfoError: null | Error;
  changePasswordConfirm: null | string;
  changePasswordError: null | Error;
  forgotConfirm: null | string;
  forgotError: null | Error;
  resetConfirm: null | string;
  resetError: null | Error;
  deleteAccountError: null | Error;
}

export interface SignUpStart {
  type: typeof UserActions.SIGN_UP_START;
  payload: UserSignUp;
}

export interface SignUpSuccess {
  type: typeof UserActions.SIGN_UP_SUCCESS;
  payload: User;
}

export interface SignUpFailure {
  type: typeof UserActions.SIGN_UP_FAILURE;
  payload: Error;
}

export interface CheckUserSession {
  type: typeof UserActions.CHECK_USER_SESSION;
  payload: null;
}

export interface SetCurrentUser {
  type: typeof UserActions.SET_CURRENT_USER;
  payload: User | null;
}

export interface SignInStart {
  type: typeof UserActions.SIGN_IN_START;
  payload: UserSignIn;
}

export interface SignInSuccess {
  type: typeof UserActions.SIGN_IN_SUCCESS;
  payload: User;
}

export interface SignInFailure {
  type: typeof UserActions.SIGN_IN_FAILURE;
  payload: Error;
}

export interface SignOutStart {
  type: typeof UserActions.SIGN_OUT_START;
  payload: null;
}

export interface SignOutSuccess {
  type: typeof UserActions.SIGN_OUT_SUCCESS;
  payload: null;
}

export interface SignOutFailure {
  type: typeof UserActions.SIGN_OUT_FAILURE;
  payload: Error;
}

export interface ChangeInfoStart {
  type: typeof UserActions.CHANGE_INFO_START;
  payload: FieldsToUpdate;
}

export interface ChangeInfoSuccess {
  type: typeof UserActions.CHANGE_INFO_SUCCESS;
  payload: User;
}

export interface ChangeInfoFailure {
  type: typeof UserActions.CHANGE_INFO_FAILURE;
  payload: Error;
}

export interface ChangePasswordStart {
  type: typeof UserActions.CHANGE_PASSWORD_START;
  payload: ChangePassword;
}

export interface ChangePasswordSuccess {
  type: typeof UserActions.CHANGE_PASSWORD_SUCCESS;
  payload: string;
}

export interface ChangePasswordFailure {
  type: typeof UserActions.CHANGE_PASSWORD_FAILURE;
  payload: Error;
}

export interface ForgotPasswordStart {
  type: typeof UserActions.FORGOT_PASSWORD_START;
  payload: string;
}

export interface ForgotPasswordSuccess {
  type: typeof UserActions.FORGOT_PASSWORD_SUCCESS;
  payload: string;
}

export interface ForgotPasswordFailure {
  type: typeof UserActions.FORGOT_PASSWORD_FAILURE;
  payload: Error;
}

export interface ResetPasswordStart {
  type: typeof UserActions.RESET_PASSWORD_START;
  payload: ResetPassword;
}

export interface ResetPasswordSuccess {
  type: typeof UserActions.RESET_PASSWORD_SUCCESS;
  payload: string;
}

export interface ResetPasswordFailure {
  type: typeof UserActions.RESET_PASSWORD_FAILURE;
  payload: Error;
}

export interface DeleteAccountStart {
  type: typeof UserActions.DELETE_ACCOUNT_START;
  payload: null;
}

export interface DeleteAccountSuccess {
  type: typeof UserActions.DELETE_ACCOUNT_SUCCESS;
  payload: string;
}

export interface DeleteAccountFailure {
  type: typeof UserActions.DELETE_ACCOUNT_FAILURE;
  payload: Error;
}

export interface ClearInfoStatuses {
  type: typeof UserActions.CLEAR_INFO_STATUSES;
  payload: null;
}

export interface ClearPasswordStatuses {
  type: typeof UserActions.CLEAR_PASSWORD_STATUSES;
  payload: null;
}

export interface GetOtherUserStart {
  type: typeof UserActions.GET_OTHER_USER_START;
  payload: string;
}

export interface GetOtherUserSuccess {
  type: typeof UserActions.GET_OTHER_USER_SUCCESS;
  payload: User;
}

export interface GetOtherUserFailure {
  type: typeof UserActions.GET_OTHER_USER_FAILURE;
  payload: Error;
}

export interface ClearOtherUserInfo {
  type: typeof UserActions.CLEAR_OTHER_USER_INFO;
  payload: null;
}

export type UserActionTypes =
  | SignUpStart
  | SignUpSuccess
  | SignUpFailure
  | SignInStart
  | SignInSuccess
  | SignInFailure
  | SignOutStart
  | SignOutSuccess
  | SignOutFailure
  | SignOutFailure
  | CheckUserSession
  | SetCurrentUser
  | ChangeInfoStart
  | ChangeInfoSuccess
  | ChangeInfoFailure
  | ChangePasswordStart
  | ChangePasswordSuccess
  | ChangePasswordFailure
  | ForgotPasswordStart
  | ForgotPasswordSuccess
  | ForgotPasswordFailure
  | ResetPasswordStart
  | ResetPasswordSuccess
  | ResetPasswordFailure
  | DeleteAccountStart
  | DeleteAccountSuccess
  | DeleteAccountFailure
  | ClearInfoStatuses
  | ClearPasswordStatuses
  | GetOtherUserStart
  | GetOtherUserSuccess
  | GetOtherUserFailure
  | ClearOtherUserInfo;
