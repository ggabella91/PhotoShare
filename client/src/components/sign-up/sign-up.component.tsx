import React, { useState, useEffect } from 'react';

import { Dispatch } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { FormInput } from '../form-input/form-input.component';
import Button from '../button/button.component';
import Alert from 'react-bootstrap/Alert';

import { AppState } from '../../redux/root-reducer';
import { Error } from '../../redux/user/user.types';
import { signUpStart } from '../../redux/user/user.actions';
import { selectUserSignUpError } from '../../redux/user/user.selectors';
import { UserSignUp, UserPayload } from '../../redux/user/user.types';

interface SignUpProps {
  signUpStart: typeof signUpStart;
  signUpError: UserPayload;
}

export const SignUp: React.FC<SignUpProps> = ({ signUpStart, signUpError }) => {
  const [userCredentials, setUserCredentials] = useState({
    name: '',
    email: '',
    password: '',
    passwordConfirm: '',
  });

  const [show, setShow] = useState(true);

  const { name, email, password, passwordConfirm } = userCredentials;

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    signUpStart({ name, email, password, passwordConfirm });
  };

  const [error, setError] = useState(false);

  useEffect(() => {
    if (signUpError) {
      setError(true);
    }
  }, [signUpError]);

  const handleRenderAlert = (type: string, message: string) => {
    if (show) {
      return (
        <Alert variant={type} onClose={() => setShow(false)} dismissible>
          {message}
        </Alert>
      );
    }
  };

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value, name } = event.target;

    setUserCredentials({ ...userCredentials, [name]: value });
  };

  return (
    <div>
      <h4 className='title'>Don't have an account yet?</h4>
      <div className='sub-text'>
        <span>Sign up below!</span>
      </div>
      <form className='sign-up-form' onSubmit={handleSubmit}>
        <FormInput
          type='text'
          name='name'
          value={name}
          onChange={handleChange}
          label='name'
        />
        <FormInput
          type='email'
          name='email'
          value={email}
          onChange={handleChange}
          label='email'
        />
        <FormInput
          type='password'
          name='password'
          value={password}
          onChange={handleChange}
          label='password'
        />
        <FormInput
          type='password'
          name='passwordConfirm'
          value={passwordConfirm}
          onChange={handleChange}
          label='confirm password'
        />
        <div>
          {error
            ? handleRenderAlert('danger', 'Error signing up. Please try again.')
            : null}
        </div>
        <div className='button'>
          <Button className='submit-button' onClick={handleSubmit}>
            Sign Up
          </Button>
        </div>
      </form>
    </div>
  );
};

interface LinkStateProps {
  signUpError: Error | null;
}

const mapStateToProps = createStructuredSelector<AppState, LinkStateProps>({
  signUpError: selectUserSignUpError,
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  signUpStart: ({ name, email, password, passwordConfirm }: UserSignUp) =>
    dispatch(signUpStart({ name, email, password, passwordConfirm })),
});

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);
