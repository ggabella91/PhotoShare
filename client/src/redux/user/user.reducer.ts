import { Reducer } from 'redux';
import { UserActions, UserActionTypes, UserState } from './user.types';

const INITIAL_STATE: UserState = {
  currentUser: null,
  signUpError: null,
  signInOrOutError: null,
};

const userReducer = (state = INITIAL_STATE, action: UserActionTypes) => {
  switch (action.type) {
    case UserActions.SET_CURRENT_USER:
      return {
        ...state,
        currentUser: action.payload,
      };
    case UserActions.SIGN_UP_SUCCESS:
      return {
        ...state,
        currentUser: action.payload,
        signUpError: null,
      };
    case UserActions.SIGN_IN_SUCCESS:
      return {
        ...state,
        currentUser: action.payload,
        signInOrOutError: null,
      };
    case UserActions.SIGN_OUT_SUCCESS:
      return {
        ...state,
        currentUser: action.payload,
        signInOrOutError: null,
      };
    case UserActions.SIGN_UP_FAILURE:
      return {
        ...state,
        signUpError: action.payload,
      };
    case UserActions.SIGN_IN_FAILURE:
    case UserActions.SIGN_OUT_FAILURE:
      return {
        ...state,
        signInOrOutError: action.payload,
      };
    default:
      return state;
  }
};

export default userReducer;
