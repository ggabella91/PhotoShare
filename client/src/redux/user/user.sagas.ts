import { takeLatest, put, all, call } from 'redux-saga/effects';
import { SagaIterator } from '@redux-saga/core';
import {ActionPattern} from '@redux-saga/types'

import { UserSignUp, User, UserActions, UserActionTypes } from './user.types';

import axios from 'axios';

import {
  setCurrentUser,
  signInSuccess,
  signInFailure,
  signOutSuccess,
  signOutFailure,
  signUpSuccess,
  signUpFailure,
} from './user.actions';

export function* signUp({
  payload: { name, email, password, passwordConfirm },
}: {
  UserActionTypes;
}): SagaIterator {
  try {
    const { data } = yield axios.post('api/v1/users/signup', {
      name,
      email,
      password,
      passwordConfirm,
    });

    yield put(signUpSuccess(data.data.user));
  } catch (err) {
    yield put(signUpFailure(err));
  }
}

export function* onSignUpStart(): SagaIterator {
  yield takeLatest<ActionPattern>(UserActions.SIGN_UP_START, signUp);
}

export function* onSignUpSuccess(): SagaIterator {
  yield takeLatest(UserActions.SIGN_UP_SUCCESS, signInAfterSignUp);
}

export function* signInAfterSignUp({ payload: user }): SagaIterator {
  yield put(setCurrentUser(user));
}

export function* onSignInStart(): SagaIterator {
  yield takeLatest(UserActions.SIGN_IN_START, signIn);
}

export function* onCheckUserSession(): SagaIterator {
  yield takeLatest(UserActions.CHECK_USER_SESSION, isLoggedIn);
}

export function* onSignOutStart(): SagaIterator {
  yield takeLatest(UserActions.SIGN_OUT_START, signOut);
}

export function* userSagas(): SagaIterator {
  yield all([]);
}
