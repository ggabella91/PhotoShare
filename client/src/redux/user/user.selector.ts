import { createSelector } from 'reselect';

import { UserState } from './user.reducer';

const selectUser = (state: any) => state.user;

export const selectCurrentUser = createSelector(
  [selectUser],
  (user: UserState) => user.currentUser
);

export const selectUserSignUpError = createSelector(
  [selectUser],
  (user: UserState) => user.signUpError
);

export const selectUserSignInOrOutError = createSelector(
  [selectUser],
  (user: UserState) => user.signInOrOutError
);
