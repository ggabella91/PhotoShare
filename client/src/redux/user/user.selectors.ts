import { createSelector } from 'reselect';

import { AppState } from '../root-reducer';
import { UserState } from './user.types';

const selectUser = (state: AppState): UserState => state.user;

export const selectCurrentUser = createSelector(
  [selectUser],
  (user: UserState) => user.currentUser
);

export const selectOtherUser = createSelector(
  [selectUser],
  (user: UserState) => user.otherUser
);

export const selectFollowersInfo = createSelector(
  [selectUser],
  (user: UserState) => user.followersInfo
);

export const selectFollowingInfo = createSelector(
  [selectUser],
  (user: UserState) => user.followingInfo
);

export const selectOtherUserError = createSelector(
  [selectUser],
  (user: UserState) => user.otherUserError
);

export const selectPostReactingUsers = createSelector(
  [selectUser],
  (user: UserState) => user.postReactingUsers
);

export const selectUserSuggestions = createSelector(
  [selectUser],
  (user: UserState) => user.userSuggestions
);

export const selectUserSuggestionsConfirm = createSelector(
  [selectUser],
  (user: UserState) => user.userSuggestionsConfirm
);

export const selectUserSuggestionsError = createSelector(
  [selectUser],
  (user: UserState) => user.userSuggestionsError
);

export const selectUserSignUpError = createSelector(
  [selectUser],
  (user: UserState) => user.signUpError
);

export const selectUserSignInOrOutError = createSelector(
  [selectUser],
  (user: UserState) => user.signInOrOutError
);

export const selectChangeInfoConfirm = createSelector(
  [selectUser],
  (user: UserState) => user.changeInfoConfirm
);

export const selectChangeInfoError = createSelector(
  [selectUser],
  (user: UserState) => user.changeInfoError
);

export const selectChangePasswordConfirm = createSelector(
  [selectUser],
  (user: UserState) => user.changePasswordConfirm
);

export const selectChangePasswordError = createSelector(
  [selectUser],
  (user: UserState) => user.changePasswordError
);

export const selectForgotError = createSelector(
  [selectUser],
  (user: UserState) => user.forgotError
);

export const selectForgotConfirm = createSelector(
  [selectUser],
  (user: UserState) => user.forgotConfirm
);

export const selectResetError = createSelector(
  [selectUser],
  (user: UserState) => user.resetError
);

export const selectResetConfirm = createSelector(
  [selectUser],
  (user: UserState) => user.resetConfirm
);

export const selectIsCurrentUserProfilePage = createSelector(
  [selectUser],
  (user: UserState) => user.isCurrentUserProfilePage
);
