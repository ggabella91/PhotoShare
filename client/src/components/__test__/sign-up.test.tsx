import { shallow } from 'enzyme';
import React from 'react';
import { SignUp } from '../sign-up/sign-up.component';
import { signUpStart } from '../../redux/user/user.actions';

it('renders a sign-up component', () => {
  const signUpWrapper = shallow(
    <SignUp
      signUpStart={(signUpPayload) => signUpStart(signUpPayload)}
      signUpError={null}
    />
  );

  expect(signUpWrapper).toMatchSnapshot();
});
