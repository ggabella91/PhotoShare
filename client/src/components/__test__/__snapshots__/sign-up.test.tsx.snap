// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a sign-up component 1`] = `
<div>
  <h4
    className="title"
  >
    Don't have an account yet?
  </h4>
  <div
    className="sub-text"
  >
    <span>
      Sign up below!
    </span>
  </div>
  <form
    className="sign-up-form"
    onSubmit={[Function]}
  >
    <FormInput
      label="name"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <FormInput
      label="email"
      name="email"
      onChange={[Function]}
      type="email"
      value=""
    />
    <FormInput
      label="password"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <FormInput
      label="confirm password"
      name="passwordConfirm"
      onChange={[Function]}
      type="password"
      value=""
    />
    <div />
    <div
      className="button"
    >
      <Button
        className="submit-button"
        onClick={[Function]}
      >
        Sign Up
      </Button>
    </div>
  </form>
</div>
`;
